### PR TITLE
Automatic threshold for valley delineation

### PR DESCRIPTION
**Pull request by fnattino**
Following up the work from @rogerkuou in #14, I have implemented a method similar to the one described in the following article to automatically set the threshold on the accumulated cost distance to define the river valley boundary: https://hess.copernicus.org/preprints/hess-2020-361/

 The only parameter now is the size of the buffer employed as basin to calculate the mean of the accumulated cost distance (in the article above they have employed previous knowledge on the wetland geometries). Testing buffer distance values from 1 to 5 km changed the mean accumulated distance from ~4 to ~16 m, so the approach seems to be relatively stable.

### CODE REVIEW

File: notebooks/river-valley/river-valley_bucharest.qmd [hunk 1]
```diff
@@ -7,74 +7,43 @@ format: html
...
 # River valley delineation for Bucharest waterways
 
 Load input files: raster DEM and multilinestrings representing the waterway:
 
 ```{r}
-dem <- rast("data/DEM_Bucharest.tif")
-river <- st_read("data/CRiSp-data-bucharest.gpkg", layer = "URC-D_dambovita_l_2017") |>
-    st_zm(drop = TRUE, what = "ZM")  # drop Z dimension
+dem <- rast("data/DEM_Bucharest_expanded.tif")
+river_geoms <- st_read("data/generated/waterway_Dâmbovița.gpkg")
```
Comment (cforgaci) [inline notebooks/river-valley/river-valley_bucharest.qmd:19]: I am missing this file from the data directory.

**Review by cforgaci — COMMENTED** [review]
Thanks @fnattino for this PR. I cannot see one of the gpkg files you are using in the notebook. See my comment below. Can you please check?

**Review by cforgaci — COMMENTED** [review]
Thanks @fnattino, looks good! 

I notice this upstream:
image
image

The valley edge overflows on the right bank of the river. This seems to be because the river line goes on top of the ridge.Is there any way we can deal with such a situation? For instance, can we add a 1-pixel buffer around the river line with minimum cell value from either side of it?

**Comment by fnattino** [comment]
> The valley edge overflows on the right bank of the river. This seems to be because the river line goes on top of the ridge.Is there any way we can deal with such a situation? For instance, can we add a 1-pixel buffer around the river line with minimum cell value from either side of it?

We had a look at the issue, and it is indeed due to the ridge laying very close to the river. Smoothening the DEM is a bit problematic there, but removing the smoothening (or applying in a following step on the computed cost distance surface) is more problematic downstreams. We thus leave the situation as it is, noting the need for improvement here https://github.com/CityRiverSpaces/prototyping/issues/1#issuecomment-2367641430 .

**Pull request merged**